pilot/pkg/security/authn/v1beta1/policy_applier.go

```diff
@@ -257,9 +266,46 @@ func convertToEnvoyJwtConfig(jwtRules []*v1beta1.JWTRule) *envoy_jwt.JwtAuthenti
 			})
 		}
 		provider.FromParams = jwtRule.FromParams
-
 		jwtPubKey := jwtRule.Jwks
-		if jwtPubKey == "" {
+		u, _ := url.Parse(jwtRule.JwksUri)
```

Comment:
can we structure the code like the following:

```go
if features.EnableRemoteJwks && jwtRule.JwksUri != "" {
  // Use remote jwks if jwksUri is non empty. Parse the jwksUri to get the cluster name, generate the jwt filter config using remoteJwks
  // If failed to parse the cluster name, fallback to let istiod to fetch the jwksUri
  // TODO: implement the logic to auto-generate the cluster so that when the flag is enabled, it will always let envoy to fetch the jwks for consistent behavior
} else {
  // Use inline jwks as existing flow, either jwtRule.jwks is non empty or let istiod to fetch the jwtRule.jwksUri
}
```

Reply:
Done